crates/core/src/host/wasmtime/wasm_instance_env.rs

```diff
@@ -582,7 +582,7 @@ impl WasmInstanceEnv {
             let data = env
                 .instance_env
                 .iter_by_col_eq(&ctx, table_id.into(), col_id.into(), value)?;
-
+            drop(ctx);
```

Comment:
We probably don't want to be dropping this execution context for every iter_by_col_eq invocation since we could have many of those in a single transaction.

Instead of passing this to iter_by_col_eq, I think we should probably add an ExecutionContext to `TxSlot` and have iter_by_col_eq just mutate it on the TxSlot directly. That way we only drop it once and therefore only write the metrics once per transaction.

Reply:
Make sense, will do that.

Reply:
I tried this, but adding`ExecutionContext` to `TxSlot` is not very straightforward, it either requires to make TxSlot use a explicit lifetime or make `ExecutionContext` to not use lifetime.
I am making this change but either of this will be a significant change `ExecutionContext` thay can/should go with different PR.

Reply:
I think your suggestion to remove the lifetime parameter from ExecutionContext is the right approach.

Reply:
`drop`, here now if dropping of Gaurd not actual memory.